satellite/overlay/selectednodescache_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package overlay_test
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"storj.io/common/memory"
+	"storj.io/common/pb"
+	"storj.io/common/storj"
+	"storj.io/common/testcontext"
+	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/overlay"
+	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+)
+
+var nodeCfg = overlay.NodeSelectionConfig{
+	AuditCount:       0,
+	UptimeCount:      0,
+	NewNodeFraction:  0.2,
+	MinimumVersion:   "v1.0.0",
+	OnlineWindow:     4 * time.Hour,
+	DistinctIP:       true,
+	MinimumDiskSpace: 100 * memory.MiB,
+}
+
+func TestInit(t *testing.T) {
+	satellitedbtest.Run(t, func(ctx *testcontext.Context, t *testing.T, db satellite.DB) {
+		cache := overlay.NewSelectedNodesCache(ctx, zap.NewNop(),
+			db.OverlayCache(), time.Hour,
+			nodeCfg,
+		)
+		// the cache should have no nodes to start
+		err := cache.Init(ctx)
+		require.NoError(t, err)
+		reputable, new := cache.Size(ctx)
+		require.Equal(t, 0, reputable)
+		require.Equal(t, 0, new)
+
+		// add some nodes to the database
+		const nodeCount = 2
+		addNodesToNodesTable(ctx, t, db.OverlayCache(), nodeCount)
+
+		// confirm nodes are in the cache once initialized
+		err = cache.Init(ctx)
+		require.NoError(t, err)
+		reputable, new = cache.Size(ctx)
+		require.Equal(t, 2, reputable)
+		require.Equal(t, 0, new)
+	})
+}
+
+func TestRefresh(t *testing.T) {
```

Comment:
Add test for:
* concurrent GetNodes.
* concurrent GetNodes with low staleness.
* concurrent Refresh